### PR TITLE
Introduce enable live metrics to distro

### PR DESCRIPTION
**Pull request by lzchen**
Add `enable_live_metrics` configuration to distro. This is going to be using an opt-in model since Python SDKs have never offered live metrics as a feature before.

Live metrics (or otherwise known as QuickPulse Service), allows for telemetry collection for metrics and performance counters in real time. The endpoint uses a similar endpoint to that of regular Application Insights endpoint.

Read more here: https://learn.microsoft.com/en-us/azure/azure-monitor/app/live-stream

### CODE REVIEW

File: sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_utils/configurations.py [hunk 1]
```diff
@@ -151,6 +153,11 @@ def _default_span_processors(configurations):
         configurations[SPAN_PROCESSORS_ARG] = []
 
 
+def _default_enable_live_metrics(configurations):
+    if ENABLE_LIVE_METRICS_ARG not in configurations:
```
Comment (johanste) [inline sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_utils/configurations.py:157]: why not simply use `dict.setdefault`?

File: sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py [hunk 2]
```diff
@@ -119,6 +132,9 @@ def _setup_tracing(configurations: Dict[str, ConfigurationValue]):
     set_tracer_provider(tracer_provider)
     for span_processor in configurations[SPAN_PROCESSORS_ARG]: # type: ignore
         get_tracer_provider().add_span_processor(span_processor) # type: ignore
+    if configurations.get(ENABLE_LIVE_METRICS_ARG):
+        qsp = _QuickpulseSpanProcessor()
+        get_tracer_provider().add_span_processor(qsp) # type: ignore
```
Comment (johanste) [inline sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py:137]: Why do we need to `type: ignore` this?

Comment (lzchen) [reply sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py]: Removed.

File: sdk/monitor/azure-monitor-opentelemetry/README.md [hunk 3]
```diff
@@ -60,6 +60,7 @@ You can use `configure_azure_monitor` to set up instrumentation for your app to
 | Parameter | Description | Environment Variable |
 |-------------------|----------------------------------------------------|----------------------|
 | `connection_string` | The [connection string][connection_string_doc] for your Application Insights resource. The connection string will be automatically populated from the `APPLICATIONINSIGHTS_CONNECTION_STRING` environment variable if not explicitly passed in. | `APPLICATIONINSIGHTS_CONNECTION_STRING` |
+| `enable_live_metrics` | Enable [live metrics][application_insights_live_metrics] feature. | `N/A` |
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry/README.md:63]: Add default

File: sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py [hunk 4]
```diff
@@ -153,6 +172,10 @@ def _setup_metrics(configurations: Dict[str, ConfigurationValue]):
     set_meter_provider(meter_provider)
 
 
+def _setup_live_metrics(configurations):
+    enable_live_metrics(**configurations)
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py:176]: Nit: If we don't expect anything else in this function in the future, let's just call enable_live_metrics at the higher level

File: sdk/monitor/azure-monitor-opentelemetry/samples/metrics/live.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        "service.instance.id": "qp_instance_id",
+    }),
+    logger_name=__name__,
+    enable_live_metrics=True,  # Enable live metrics configuration
+)
+
+tracer = trace.get_tracer(__name__)
+logger = logging.getLogger(__name__)
+
+while True:
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry/samples/metrics/live.py:28]: Nit: Could use an explaining comment

**Review by jeremydvoss — APPROVED** [review]

**Pull request merged**